server/game-service/main.go

```diff
@@ -0,0 +1,343 @@
+	Players    []*websocket.Conn // tracks the players in a game
+	Scores     []int             // tracks the scores of both players
+	Turn       int
+	Timer      int    // tracks the time left in a game
+	Difficulty string // tracks the difficulty of a game
+	GameOver   bool   // tracks if a game is over
+	Active     bool   // tracks if a game is active
+	Round      int    // tracks the number of rounds in a game
+}
+
+// This section is for the global variables
+var (
+	games    = make(map[string]*Game) // store active games
+	gamesMux sync.Mutex
+	upgrader = websocket.Upgrader{
+		CheckOrigin: func(r *http.Request) bool { return true },
+	}
+)
+
+// This function is responsible for starting the game loop
+func startGame(gameID string) {
+	gamesMux.Lock()
+	game, exists := games[gameID]
+	if !exists {
+		gamesMux.Unlock()
+		fmt.Println("[ERROR] Game not found:", gameID)
+		return
+	}
+	gamesMux.Unlock()
+
+	rand.Seed(time.Now().UnixNano())
+	game.Turn = rand.Intn(2) // this is for randomly selecting the starting player
+	game.Round = 1           // the round will start in 1
+
+	// This section is for the difficulty levels
+	durations := map[string]int{"easy": 180, "medium": 420, "hard": 600}
+	game.Timer = durations[game.Difficulty]
+	game.Active = true
+
+	fmt.Printf("[GAME START] Game ID: %s | Difficulty: %s | Timer: %d seconds | Player %d starts (Round %d)\n",
+		gameID, game.Difficulty, game.Timer, game.Turn, game.Round)
+
+	// This function is responsible for starting the game loop in a new goroutine
+	go func() {
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
+
+		for game.Timer > 0 {
+			<-ticker.C
+
+			game.Mutex.Lock()
+			if game.GameOver {
+				game.Mutex.Unlock()
+				fmt.Printf("[GAME OVER] Game ID: %s\n", gameID)
+				return
+			}
+			game.Timer--
+			fmt.Printf("[TIMER] Game ID: %s | Time Left: %d seconds | Round: %d | Scores: [%d, %d]\n",
+				gameID, game.Timer, game.Round, game.Scores[0], game.Scores[1])
+			game.broadcastState()
+			game.Mutex.Unlock()
+		}
+
+		// This section means the game is over
+		game.Mutex.Lock()
+		game.GameOver = true
+		game.broadcastState()
+		game.Mutex.Unlock()
+
+		// This section determines the winner of the game, and tie if there are no winners
+		winner := -1
+		if game.Scores[0] > game.Scores[1] {
+			winner = 0
+		} else if game.Scores[1] > game.Scores[0] {
+			winner = 1
+		}
+
+		// This section broadcasts the game over message with the winner
+		game.Mutex.Lock()
+		state := map[string]interface{}{
+			"scores":   game.Scores,
+			"turn":     game.Turn,
+			"timer":    game.Timer,
+			"gameOver": game.GameOver,
+			"round":    game.Round,
+			"winner":   winner,
+		}
+		message, _ := json.Marshal(state)
+		for i, player := range game.Players {
+			if player != nil {
+				err := player.WriteMessage(websocket.TextMessage, message)
+				if err != nil {
+					fmt.Printf("[ERROR] Failed to send state to Player %d | Error: %v\n", i, err)
+				}
+			}
+		}
```

Comment:
Use guard clauses to avoid deeply nested if statements

Reply:
```suggestion
		for i, player := range game.Players {
			if player != nil {
				err := player.WriteMessage(websocket.TextMessage, message)
                                 break
			}
			if err != nil {
				fmt.Printf("[ERROR] Failed to send state to Player %d | Error: %v\n", i, err)
			}
		}
```

I would suggest doing it like this